### ISSUE TITLE
Upstream cross-compilation support

### ISSUE THREAD

**Issue by alyssais** [nix-ocaml/nix-overlays#698]
It would be really great if Nixpkgs could cross-compile OCaml out of the box.  What would need to happen to get the cross-compilation support from this overlay upstream?  I would be willing to put work into this, but if you already have thought about what would need to happen it would be great to hear from you.

**Comment by anmonteiro** [comment]
I agree this would be desirable to upstream. I haven't made many strides towards that however. Here's the current state of cross compilation support in this repo:

- changes live in the [`cross/` folder](https://github.com/nix-ocaml/nix-overlays/tree/master/cross)
- the OCaml compiler needs a small patch + a custom build process which you can find [here](https://github.com/nix-ocaml/nix-overlays/blob/master/cross/ocaml-compiler.nix)
- then we need to make the OCaml package set (esp. findlib) aware of the 2 toolchains (`buildPackages` and the cross-compilation environment). This is all done in [`cross/ocaml.nix`](https://github.com/nix-ocaml/nix-overlays/blob/master/cross/ocaml.nix)
  - IMO this is where most of the complexity lies: nixpkgs uses `OCAMLPATH` to tell findlib about OCaml packages, but we need to actually write down `findlib.conf` files for both the build and cross toolchains.
